### DISCUSSION TITLE
Problem with Windows bridges

### DISCUSSION THREAD

**Discussion by raven-kg** [falkTX/Carla#1571] [Q&A]
Hi there!
I have a problem with my carla installation

> $ carla --version
> Using Carla version 2.4.1
>   Python version: 3.6.8
>   Qt version:     5.15.2
>   PyQt version:   5.15.0
>   Binary dir:     /usr/lib64/carla
>   Resources dir:  /usr/share/carla/resources
> 

it has been built using the following steps:

> $ make features
> ---> Main features
> Front-End:     YES
> LV2 plugin:    YES
> VST2 plugin:   YES
> Link support:  YES
> OSC support:   YES
> Binary detect: NO  [Missing libmagic/file]
> ---> Engine drivers
> JACK:        YES
> ALSA:        YES
> PulseAudio:  YES
> CoreAudio:   NO  [MacOS only]
> ASIO:        NO  [Windows only]
> DirectSound: NO  [Windows only]
> WASAPI:      NO  [Windows only]
> ---> Plugin formats:
> Internal: YES
> LADSPA:   YES
> DSSI:     YES
> LV2:      YES
> VST2:     YES (with UI)
> VST3:     YES (with UI, using JUCE)
> AU:       NO  [MacOS only]
> ---> LV2 UI toolkit support:
> External: YES (direct)
> Gtk2:     YES (bridge)
> Gtk3:     YES (bridge)
> Qt4:      NO  [Qt4 missing]
> Qt5:      YES (bridge)
> X11:      YES (direct+bridge)
> Cocoa:    NO  [MacOS only]
> Windows:  NO  [Windows only]
> ---> File formats:
> Basic: YES
> MP3:   YES
> Extra: NO [FFmpeg missing or too new]
> SF2/3: YES
> SFZ:   YES
> ---> Internal plugins:
> Basic Plugins:    YES
> Carla-Patchbay:   YES
> Carla-Rack:       YES
> External Plugins: NO
> 
> $ /usr/bin/make -O -j4 SKIP_STRIPPING=true NOOPT=true V=1
> ...
> 
> $ WINECC=/usr/bin/winegcc
> $ make posix32 wine32
> ...
> $ env CFLAGS= CPPFLAGS= CXXFLAGS= LDFLAGS= make win32 CC=i686-w64-mingw32-gcc CXX=i686-w64-mingw32-g++
> ...
> $  make posix64 wine64
> ...
> $ env CFLAGS= CPPFLAGS= CXXFLAGS= LDFLAGS= make win64 CC=x86_64-w64-mingw32-gcc CXX=x86_64-w64-mingw32-g++
> ...
> $ sudo make install 'INSTALL=/usr/bin/install -p' PREFIX=/usr LIBDIR=/usr/lib64
> 

Carla itself run without any problem, but wine* bridges do not appear in the Available tools list (as well as posix ones) even though the files exist.
https://i.imgur.com/OYO7AqW.png

>$ ls /usr/lib64/carla/ | grep discovery
> carla-discovery-native
> carla-discovery-posix32
> carla-discovery-posix64
> carla-discovery-win32.exe
> carla-discovery-win64.exe

What could be wrong with my setup?

**Comment by falkTX** [comment]
I am not sure if carla installs those tools you need, check if they are in ./bin/ during build.

**Reply by raven-kg** [reply]
Do you mean some of these binaries?

> $ ls ./bin | grep carla
> carla-bridge-lv2-gtk2
> carla-bridge-lv2-gtk3
> carla-bridge-lv2-modgui
> carla-bridge-lv2-qt5
> carla-bridge-lv2.so
> carla-bridge-lv2-x11
> carla-bridge-native
> carla-bridge-posix32
> carla-bridge-posix64
> carla-bridge-win32.exe
> carla-bridge-win64.exe
> carla-discovery-native
> carla-discovery-posix32
> carla-discovery-posix64
> carla-discovery-win32.exe
> carla-discovery-win64.exe

**Reply by falkTX** [reply]
the wine stuff is missing, dont recall the exact name now but it is "jackbridge" something with *.dll extension
somehow you dont have those, but they should be there assuming you ran "make wine64"

**Reply by raven-kg** [reply]
Ah, the dlls exists, I just removed them from the list to make it shorter. Here is entire directory content:

> $ ls ./bin
> carla-bridge-lv2-gtk2
> carla-bridge-lv2-x11
> carla-bridge-win64.exe
> carla-discovery-win64.exe
> CarlaPatchbay64.so
> CarlaVstFxShell.so
> libcarla_host-plugin.so
> libcarla_standalone2.so
> carla-bridge-lv2-gtk3
> carla-bridge-native
> carla-discovery-native
> carla.lv2
> CarlaPatchbayFX.so
> CarlaVstShell.so
> libcarla_interposer-jack-x11.so
> libcarla_utils.so
> carla-bridge-lv2-modgui
> carla-bridge-posix32
> carla-discovery-posix32
> carla-lv2-export
> CarlaPatchbay.so
> jack
> libcarla_interposer-safe.so
> resources
> carla-bridge-lv2-qt5
> carla-bridge-posix64
> carla-discovery-posix64    
> CarlaPatchbay16.so         
> CarlaRackFX.so      
> jackbridge-wine32.dll  
> libcarla_interposer-x11.so       
> styles
> carla-bridge-lv2.so      
> carla-bridge-win32.exe  
> carla-discovery-win32.exe  
> CarlaPatchbay32.so         
> CarlaRack.so        
> jackbridge-wine64.dll  
> libcarla_native-plugin.so

**Reply by falkTX** [reply]
ok then all is fine.
did you enable the experimental settings?

**Reply by raven-kg** [reply]
Oh, it works! Yeah, I didn't enabled experimental settings and wine bridges, so that was the reason. Thanks for helping with that!
